### PR TITLE
Task/improve session and error handling

### PR DESCRIPTION
**Pull request by dlozlla**

### CODE REVIEW

File: lib/routes/pages.js [hunk 1]
```diff
@@ -64,10 +64,14 @@ function joinOrga(req, res) {
     type: 'oauth',
     token: req.user.accessToken
   });
+
+  logger.debug('Github user ' + req.user.username + ' with AccessToken ' + req.user.accessToken +
+      ' is requesting to join');
+
   github.user.getEmails({}, function(userErr, userRes) {
     if (userErr) {
       logger.error('An unexpected error occurred while using github API to get github user ' +
```
Comment (jlorgal) [inline lib/routes/pages.js]: You should use %s instead of string concatenation

File: lib/routes/pages.js [hunk 2]
```diff
@@ -64,10 +64,14 @@ function joinOrga(req, res) {
     type: 'oauth',
     token: req.user.accessToken
   });
+
+  logger.debug('Github user ' + req.user.username + ' with AccessToken ' + req.user.accessToken +
```
Comment (jlorgal) [inline lib/routes/pages.js]: You should use %s instead of string concatenation

**Comment by labajo** [comment]
LGTM

**Comment by jlorgal** [comment]
LGTM

**Pull request merged**